src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -325,10 +310,17 @@ internal static bool ValidateParentTagName(string parentTag, ErrorSink errorSink
                     errorSink: errorSink);
         }
 
-        private static bool ValidateName(
-            string name,
-            bool targetingAttributes,
-            ErrorSink errorSink)
+        private static bool TryGetRequiredAttributeDescriptors(
+            string requiredAttributes,
+            ErrorSink errorSink,
+            out IEnumerable<TagHelperRequiredAttributeDescriptor> descriptors)
+        {
+            var parser = new RequiredAttributeParser(requiredAttributes);
```

Comment:
:-1: 
To make this stateless I'd need to pass around a lot more information in method signatures. Wayyy not worth it. It's already complex that we have a separate parser for required attributes.